src/services/photos/database/photosRealmDB.ts

```diff
@@ -0,0 +1,170 @@
+import { BaseLogger } from '@internxt-mobile/services/common';
+import { fs } from '@internxt-mobile/services/FileSystemService';
+import { Photo, PhotoStatus } from '@internxt/sdk/dist/photos';
+import { Asset } from 'expo-media-library';
+import Realm, { ObjectSchema } from 'realm';
+
+// Store an alreay synced Photo
+const PhotoSchema: ObjectSchema = {
+  primaryKey: 'photoId',
+  properties: {
+    name: { type: 'string', indexed: true },
+    photoId: { type: 'string', indexed: true },
+    takenAt: 'date',
+    hash: 'string',
+    photoJSON: 'string',
+    status: 'string',
+  },
+  name: 'Photo',
+};
+
+type PhotoDoc = {
+  name: string;
+  photoId: string;
+  takenAt: Date;
+  hash: string;
+  status: PhotoStatus;
+  photoJSON: string;
+};
+
+// Store a DevicePhoto
+const DevicePhotoSchema: ObjectSchema = {
+  primaryKey: 'name',
+  properties: {
+    name: { type: 'string', indexed: true },
+    takenAt: 'date',
+    devicePhotoJSON: 'string',
+  },
+  name: 'DevicePhoto',
+};
+
+type DevicePhotoDoc = {
+  name: string;
+  takenAt: Date;
+  devicePhotoJSON: string;
+};
+export class PhotosRealmDB {
+  private _realm: Realm | null = null;
+  private logger: BaseLogger = new BaseLogger({
+    tag: 'PHOTOS_REALM_DB',
+    enabled: false,
+  });
+
+  async init() {
+    this._realm = await Realm.open({
+      schema: [PhotoSchema],
+      deleteRealmIfMigrationNeeded: true,
+    });
+
+    this.logger.info('Photos Realm DB is ready');
+  }
+
+  clear() {
+    if (!this._realm) return;
+    this.realm.write(() => {
+      this.realm.deleteAll();
+    });
+
+    fs.unlinkIfExists(this.realm.path);
+  }
+
+  private get realm() {
+    if (!this._realm) throw new Error('No connection open, call init() first');
+
+    return this._realm as Realm;
+  }
+
+  public async saveDevicePhotos(devicePhotos: Asset[]) {
```

Comment:
`async` is not required here. Also, I suppose that you are not waiting for the write operation to finish here, intentionally.

Reply:
actually RealmDB is sync, so the async should be removed

Reply:
Go then